integration/test/Test/Conversation.hs

```diff
@@ -177,3 +179,136 @@ testCreateConversationNonFullyConnected = do
+
+        -- assert conversation deleted from domainA
+        retryT $
+          bindResponse (getConversation uA convId) $ \r ->
+            r.status `shouldMatchInt` 404
+
+        -- assert federation.delete event is sent twice
+        void $ awaitNMatches 2 3 (\n -> nPayload n %. "type" `isEqual` "federation.delete") ws
+
+      -- assert no conversation.delete event is sent to uA
+      eventPayloads <-
+        getNotifications uA "cA" def
+          >>= getJSON 200
+          >>= \n -> n %. "notifications" & asList >>= \ns -> for ns nPayload
+
+      forM_ eventPayloads $ \p ->
+        p %. "type" `shouldNotMatch` "conversation.delete"
+
+testDefederationOneOnOne :: HasCallStack => App ()
+testDefederationOneOnOne = do
+  let setFederationConfig =
+        setField "optSettings.setFederationStrategy" "allowDynamic"
+          >=> removeField "optSettings.setFederationDomainConfigs"
+          >=> setField "optSettings.setFederationDomainConfigsUpdateFreq" (Aeson.Number 1)
+  startDynamicBackends
+    [ def {dbBrig = setFederationConfig},
+      def {dbBrig = setFederationConfig}
+    ]
+    $ \dynDomains -> do
+      domains@[domainA, domainB] <- pure dynDomains
+      connectAllDomainsAndWaitToSync 1 domains
+      [uA, uB] <- createAndConnectUsers [domainA, domainB]
+      -- figure out on which backend the 1:1 conversation is created
+      conn <- getConnection uA uB
+      qConvId <- conn.json %. "qualified_conversation"
+
+      -- check conversation exists and uB is a member from POV of uA
+      bindResponse (getConversation uA qConvId) $ \r -> do
+        r.status `shouldMatchInt` 200
+        members <- r.json %. "members.others" & asList
+        qIds <- for members (\m -> m %. "qualified_id")
+        uBQId <- objQidObject uB
+        qIds `shouldMatchSet` [uBQId]
+
+      -- check conversation exists and uA is a member from POV of uB
+      bindResponse (getConversation uB qConvId) $ \r -> do
+        r.status `shouldMatchInt` 200
+        members <- r.json %. "members.others" & asList
+        qIds <- for members (\m -> m %. "qualified_id")
+        uAQId <- objQidObject uA
+        qIds `shouldMatchSet` [uAQId]
+
+      conversationOwningDomain <- objDomain qConvId
+
+      when (domainA == conversationOwningDomain) $ do
+        -- conversation is created on domainA
+        assertFederationTerminatingUserNoConvDeleteEvent uB qConvId domainB domainA
+
+      when (domainB == conversationOwningDomain) $ do
+        -- conversation is created on domainB
+        assertFederationTerminatingUserNoConvDeleteEvent uA qConvId domainA domainB
+
+      when (domainA /= conversationOwningDomain && domainB /= conversationOwningDomain) $ do
+        -- this should not happen
+        assertFailure $
+          "conversation was expected to be created on "
+            <> domainA
+            <> " or "
+            <> domainB
+            <> " but was created on '"
+            <> conversationOwningDomain
+            <> "'"
```

Comment:
we don't expect this to ever happen, right?  what domain should that be?  so i wouldn't spend so much code on an elaborate error message here, but just case-switch over `conversationOwningDomain` with a catch-all saying `error "impossible"`.